tech/post/2017-08-29-Client-go-tools-cache-Informer.md

```diff
@@ -0,0 +1,144 @@
+---
+layout: post
+title: The mechanism of client-go Informer
+description: describe the mechanism of Informer including the implementation of method NewSharedIndexInformer, NewInformer and so on
+---
+
+## 1. Common Components
+
+### 1.1 threadSafeMap
+threadSafeMap stores each object and builds different type of index for it in order to access it flexibly.
+
+#### Class Diagram
+![]({{ site.url }}/images/2017-08-29-Client-go-tools-cache-Informer/client-go.tools.cache.ThreadSafeStore.svg)
```

Comment:
It's better to use the following ways to show the pictures, It is better for users to review it by md file directly.
![](/images/2017-08-29-Client-go-tools-cache-Informer/client-go.tools.cache.ThreadSafeStore.svg)
![](../images/2017-08-29-Client-go-tools-cache-Informer/client-go.tools.cache.ThreadSafeStore.svg)